### PR TITLE
Agnhost version sync

### PR DESCRIPTION
**Pull request by BenTheElder**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

I also:
- made the cobra.Command in agnhost.go multiline for easier matching on the right field
- updated the out of sync agnhost source image versions

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by BenTheElder** [comment]
these have de-synced multiple times. most recently caught in https://github.com/kubernetes/kubernetes/pull/94135#discussion_r474162724

File: build/dependencies.yaml [hunk 1]
```diff
@@ -1,4 +1,20 @@
...
+  - name: "agnhost"
+    version: "2.22"
+    refPaths:
+    - path: test/images/agnhost/VERSION
+      match: \d.\d
+    - path: test/images/agnhost/agnhost.go
+      match: "Version:"
+
+  # then after merge and successful postsubmit image push / promotion, bump this
+  - name: "agnhost: dependents"
```
Comment (BenTheElder) [inline build/dependencies.yaml:12]: I suspect there are more, but I didn't turn up any that are obviously intended to be in sync here yet.
If nothing else though, this entry should prevent an incorrect addition of this file to the entry above.

There is also the rarely updated private image from this that we literally only use to test pulling a private image (w/ auth).

**Comment by spiffxp** [comment]
This missed updating test/utils/image/manifest.go, suggested update in https://github.com/kubernetes/kubernetes/pull/89582/files#r480392573

**Pull request merged**